### PR TITLE
refactor: Replace Ushahidi\Tool\Filesystem with Flysystem from Lumen

### PR DESCRIPTION
**Pull request by rjmackay**
This pull request makes the following changes:
- Remove Ushahidi\Tool\Filesystem intermediary
- Fetch filesystem instance direct from Lumen
- Use Uploader for deleting files too
- Update CreateMedia and DeleteCSV usecases to remove Filesystem dependency
- Add filesystem config
- Add symlink from httpdocs/storage to storage/app/public
- Remove CDN config

Test checklist:
- [x] bin/phpspec run
- [x] bin/behat tests/integration/csv.feature
- [x] bin/behat tests/integration/media.feature

Ping @ushahidi/platform

### CODE REVIEW

File: src/Core/Tool/Uploader.php [hunk 1]
```diff
@@ -92,6 +92,17 @@ public function upload(UploadData $file, $filename = null)
 			]);
 	}
 
+	/**
+	 * Clean up file if upload failed
+	 * @param  $file  filepath
+	 */
+	public function delete($file)
```
Comment (rjmackay) [inline src/Core/Tool/Uploader.php]: Its plausible that now I've added this method, I should really rename this to `FileActions` or something. But I don't really care that much :)

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -101,6 +102,14 @@ protected function configureAuraServices(\Aura\Di\ContainerInterface $di)
             'clientUrl' => $di->lazyGet('clienturl')
         ]));
 
+        // Configure filesystem
+        // The Ushahidi filesystem adapter returns a flysystem adapter for a given
+        // cdn type based on the provided configuration
+        $di->set('tool.filesystem', function () {
+            // Get the underlying League\Flysystem\Filesystem instance
+            return $this->app->make('filesystem')->disk()->getDriver();
```
Comment (rjmackay) [inline app/Providers/AppServiceProvider.php]: Future music: We should define both public and private filesystem. CSV uploads could be private, but image uploads should be public.

File: config/filesystems.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    |
+    | Here you may configure as many filesystem "disks" as you wish, and you
+    | may even configure multiple disks of the same driver. Defaults have
+    | been setup for each driver as an example of the required options.
+    |
+    | Supported Drivers: "local", "ftp", "sftp", "s3", "rackspace"
+    |
+    */
+
+    'disks' => [
```
Comment (rowasc) [inline config/filesystems.php]: is rackspace not a thing anymore?

**Comment by rowasc** [comment]
@rjmackay the tests fail for me. But most tests also fail in the lumen branch. So it might be something in my setup. Not really sure.

File: config/filesystems.php (hunk 3 above)
Comment (rjmackay) [reply config/filesystems.php]: Um it is but this is just the default config file from laravel. I'll expand that with rackspace stuff. Oops

**Comment by rjmackay** [comment]
> the tests fail for me. But most tests also fail in the lumen branch. So it might be something in my setup. Not really sure.

The full suite will fail. The specific tests called out above should pass I think

**Pull request merged**